### PR TITLE
*: add methods to session.Context and refactor some code about `ddl`

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/32031

- change some mock sessions in the test to real session
- add 3 methods in sessionctx.Context about txn and disk full
- refine some codes
  - introduce a interface about job iter

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: util/mock/context.go [hunk 1]
```diff
@@ -247,6 +248,23 @@ func (c *Context) RefreshVars(ctx context.Context) error {
...
 
+// RollbackTxn indicates an expected call of RollbackTxn
+func (c *Context) RollbackTxn(ctx context.Context) {
+	defer c.sessionVars.SetInTxn(false)
+	if c.txn.Valid() {
+		terror.Log(c.txn.Rollback())
+	}
+}
+
+// CommitTxn indicates an expected call of CommitTxn
```
Comment (zimulala) [inline util/mock/context.go:259]:
```suggestion
// CommitTxn indicates an expected call of CommitTxn.
```

File: util/mock/context.go [hunk 2]
```diff
@@ -247,6 +248,23 @@ func (c *Context) RefreshVars(ctx context.Context) error {
 	return nil
 }
 
+// RollbackTxn indicates an expected call of RollbackTxn
```
Comment (zimulala) [inline util/mock/context.go:251]:
```suggestion
// RollbackTxn indicates an expected call of RollbackTxn.
```

File: meta/meta.go [hunk 3]
```diff
@@ -1040,23 +1040,28 @@ func (m *Meta) GetLastNHistoryDDLJobs(num int) ([]*model.Job, error) {
...
 		return nil, err
 	}
-	return &LastJobIterator{
+	return &HLastJobIterator{
 		iter: iter,
 	}, nil
 }
 
+// HLastJobIterator is the iterator for gets the latest history.
+type HLastJobIterator struct {
```
Comment (zimulala) [inline meta/meta.go:1059]: Why add this struct or make this change？

File: ddl/column_test.go [hunk 4]
```diff
@@ -288,22 +288,22 @@ func TestColumnBasic(t *testing.T) {
...
 
-	jobID = testCreateIndex(tk, t, testNewContext(store), tableID, false, "c5_idx", "c5", dom)
+	jobID = testCreateIndex(tk, t, testkit.NewTestKit(t, store).Session(), tableID, false, "c5_idx", "c5", dom)
 	testCheckJobDone(t, store, jobID, true)
 
-	jobID = testDropColumnInternal(tk, t, testNewContext(store), tableID, "c5", false, dom)
+	jobID = testDropColumnInternal(tk, t, testkit.NewTestKit(t, store).Session(), tableID, "c5", false, dom)
 	testCheckJobDone(t, store, jobID, false)
 
-	jobID = testDropColumnInternal(tk, t, testNewContext(store), tableID, "c6", true, dom)
```
Comment (zimulala) [inline ddl/column_test.go:306]: Could we remove the function of `testNewContext`? And why?

File: sessionctx/context.go [hunk 5]
```diff
@@ -48,7 +49,12 @@ type Context interface {
 	NewTxn(context.Context) error
 	// NewStaleTxnWithStartTS initializes a staleness transaction with the given StartTS.
 	NewStaleTxnWithStartTS(ctx context.Context, startTS uint64) error
-
+	// SetDiskFullOpt set the disk full opt when tikv disk full happened.
+	SetDiskFullOpt(level kvrpcpb.DiskFullOpt)
+	// RollbackTxn rolls back the current transaction.
+	RollbackTxn(ctx context.Context)
+	// CommitTxn commits the current transaction.
+	CommitTxn(ctx context.Context) error
```
Comment (zimulala) [inline sessionctx/context.go:57]: Please add the reason for adding these functions here to the PR description.

File: meta/meta.go (hunk 3 above)
Comment (xiongjiwei) [reply meta/meta.go:1059]: we will add another implementation(concurrent ddl implementation) for `LastJobIterator` in the future

File: ddl/column_test.go (hunk 4 above)
Comment (xiongjiwei) [reply ddl/column_test.go:306]: not now, there are still some other usage

File: sessionctx/context.go (hunk 5 above)
Comment (xiongjiwei) [reply sessionctx/context.go:57]: it already has some functions related to txn like `StmtCommit` , `StmtRollback`, and it very naturally adds `rollbackTxn` etc.

**Review by wjhuang2016 — APPROVED** [review]

File: sessionctx/context.go (hunk 5 above)
Comment (zimulala) [reply sessionctx/context.go:57]: What about the `SetDiskFullOpt`?

Comment (xiongjiwei) [reply sessionctx/context.go:57]: the `session.context` will be used in ddl job operation, and only general ddls allowed to be executed when TiKV is disk full. So we will use `SetDiskFullOpt` method to control it

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**